### ISSUE TITLE
bug: @Watch methods not firing in code build as dist-custom-elements output target

### ISSUE THREAD

**Issue by bpark** [ionic-team/stencil#4221]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

3.2.0

### Current Behavior

Code build as dist-custom-elements output, @Watch decorated methods are not executed when a watched attribute is changed dynamically.

A similar issue was reported in Jun 1, 2020 and got fixed, but as I can see the fixes got reverted afterwards, see here https://github.com/ionic-team/stencil/issues/2478

### Expected Behavior

Methods decorated with @Watch should be executed on attribute change in code generated by a dist-custom-elements output

### System Info

```shell
System: node 18.15.0
    Platform: windows (10.0.22621)
   CPU Model: AMD Ryzen 7 5800X 8-Core Processor              (16 cpus)
    Compiler: C:\dev\stencil-custom-elements-watch-bug\node_modules\@stencil\core\compiler\stencil.js
       Build: 1678807683
     Stencil: 3.2.0
  TypeScript: 4.9.5
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.16.6
```

### Steps to Reproduce

1. Clone repo https://github.com/bpark/stencil-custom-elements-watch-bug
2. Run npm install and npm run build
3. Start a web server in dist/ folder and open the index.html
4. Click either of "Change prop" buttons and observe watchCallCount=... values

### Code Reproduction URL

https://github.com/bpark/stencil-custom-elements-watch-bug

**Comment by rwaskiewicz** [comment]
Hey @bpark, 

Thanks for the detailed reproduction! I've confirmed that the method decorated with `@Watch` is not firing properly. Is this only reproducible with `externalRuntime` set to `false`?

**Comment by bpark** [comment]
Hi @rwaskiewicz ,

the problem also occurs when the `externalRuntime` flag is set to `true`. Originally I found the problem when I integrated a dist-custom-elements built WebComponent into an angular application with `defineCustomElement`.

**Comment by rwaskiewicz** [comment]
👋 

The fix for this issue has been included in the [🏒 v3.2.2 release of Stencil](https://github.com/ionic-team/stencil/releases/tag/v3.2.2), which was released earlier today. As a result, I'm going to close this issue. Should it reappear, please feel free to open a new issue. Thanks!
